src/doom/d_main.c

```diff
@@ -296,12 +312,33 @@ boolean D_Display (void)
     inhelpscreensstate = inhelpscreens;
     oldgamestate = wipegamestate = gamestate;
     
+
+    if(crispy->snowflakes)
+    {
+        for(size_t i = 0; i < 5000; i++)
+        {
+            snowflakes[i].y += Crispy_Random() % 2;
+            if(snowflakes[i].y < 0)
+                continue;
+
+            snowflakes[i].x += 1 - Crispy_Random() % 3;
+            if(snowflakes[i].y >= SCREENHEIGHT)
+                snowflakes[i].y = 0;
+            if(snowflakes[i].x >= SCREENWIDTH)
+                snowflakes[i].x = 0;
+            if(snowflakes[i].x < 0)
+                snowflakes[i].x = SCREENWIDTH;
+
+            I_VideoBuffer[snowflakes[i].x + snowflakes[i].y * SCREENWIDTH] = 4;
+        }
```

Comment:
Better determine the best matching snow flake color once during initialization at run-time. Some PWADs change the palette in unexpected ways.